crates/network-scanner/examples/mdns.rs

```diff
@@ -0,0 +1,26 @@
+use std::time::Duration;
+
+use network_scanner::{mdns, task_utils::TaskManager};
+
+#[tokio::main]
+pub async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::SubscriberBuilder::default()
+        .with_max_level(tracing::Level::INFO)
+        .with_thread_names(true)
+        .init();
+
+    let deamon = mdns_sd::ServiceDaemon::new()?;
```

Comment:
suggestion: Could you hide the `mdns_sd` dependency behind our own crate? I.e.: make sure that the user code of our `network-scanner` crate does not need to depend on `mdns_sd` directly. This means that the example should not call `mdns_sd` directly.

rationale: Easier to use API, and less dependencies to declare (and forget) in consumer crate (i.e.: `devolutions-gateway`).

Reply:
updated